### ISSUE TITLE
No events when moving from 21.3 to 21.4 / 21.5

### ISSUE THREAD

**Issue by cjd118** [getsentry/onpremise#960]
### Version Information

From 21.3 to 21.4 / 21.5

### Steps to Reproduce

We're upgrading Sentry from 9.1.2 to 21.4.

Can get as far as 21.3 without any issues and with preserving event data okay. 

However, the jump from 21.3 to 21.4 causes all event data to seemingly get removed. I say seemingly because although it shows no event data, if I go to an event URL I do see issue information but no event breakdown etc. 

The only really non-standard thing we are doing is running postgres 11 rather than 9 (somehow the base image got updated way back and it's proven hard to downgrade). 

There are no errors during install or run and I couldn't find any logs from Zookeeper that looked relevant. 

My hunch would be it has something to do with a Snuba migration 'truncate events' maybe? Which appeared in 21.4.0 - https://github.com/getsentry/snuba/blob/5903238e87cd820b7c4e490d7478aae3eab30ba4/snuba/migrations/snuba_migrations/events/0015_truncate_events.py

Otherwise I'm stumped, is there anywhere else I should look or check for? 

Thanks

Edit, the exact error message when drilling into an event (the event count is visible, the event title is visible, some event meta data is visible. But stack trace, etc. is not). 
```
"Sorry, the events for this issue could not be found.
This could be due to a handful of reasons:

The events are still processing and are on their way
The events have been deleted
There is an internal systems error or active issue"
```

Also clickhouse database seems tiny for 50k events?

```
┌─table─────────────────┬─size──────┬───min_date─┬───max_date─┐
│ metric_log            │ 7.59 MiB  │ 2021-05-14 │ 2021-05-24 │
│ migrations_local      │ 1.32 KiB  │ 0000-00-00 │ 0000-00-00 │
│ trace_log_0           │ 1013.00 B │ 2021-05-14 │ 2021-05-14 │
│ sentry_local          │ 3.67 MiB  │ 0000-00-00 │ 0000-00-00 │
│ errors_local          │ 22.83 KiB │ 0000-00-00 │ 0000-00-00 │
│ outcomes_raw_local    │ 2.59 KiB  │ 0000-00-00 │ 0000-00-00 │
│ outcomes_hourly_local │ 1.81 KiB  │ 0000-00-00 │ 0000-00-00 │
└───────────────────────┴───────────┴────────────┴────────────┘
```

**Comment by BYK** [comment]
Can you share the install logs please, something with your Clickhouse data seems very off.

**Comment by fpacifici** [comment]
Could you also provide the output of this clickhouse query? `SELECT * FROM migrations_local`. THat will tell us exactly which migrations did run and which did not. 
Asking because, if the list of tables you provided is the complete list of the tables on your DB, that is missing several tables that should have been created long before 21.3 (like `transacitons_local`, `groupedmessage_local`, `groupassignee_local` and `sessions_raw_local`)

**Comment by cjd118** [comment]
Hi,

Put various bits of info below.

Here is Clickhouse DB on 21.3:
```
┌─table─────────────────┬─size──────┬───min_date─┬────ax_date─┐
│ metric_log            │ 9.57 MiB  │ 2021-05-14 │ 2021-05-25 │
│ migrations_local      │ 1.32 KiB  │ 0000-00-00 │ 0000-00-00 │
│ trace_log_0           │ 1013.00 B │ 2021-05-14 │ 2021-05-14 │
│ sentry_local          │ 3.67 MiB  │ 0000-00-00 │ 0000-00-00 │
│ errors_local          │ 22.83 KiB │ 0000-00-00 │ 0000-00-00 │
│ outcomes_raw_local    │ 2.59 KiB  │ 0000-00-00 │ 0000-00-00 │
│ outcomes_hourly_local │ 1.81 KiB  │ 0000-00-00 │ 0000-00-00 │
└───────────────────────┴───────────┴────────────┴────────────┘
```
Here it is on 21.4, no change
```
┌─table─────────────────┬─size──────┬───min_date─┬───max_date─┐
│ metric_log            │ 9.60 MiB  │ 2021-05-14 │ 2021-05-25 │
│ migrations_local      │ 1.32 KiB  │ 0000-00-00 │ 0000-00-00 │
│ trace_log_0           │ 1013.00 B │ 2021-05-14 │ 2021-05-14 │
│ sentry_local          │ 3.67 MiB  │ 0000-00-00 │ 0000-00-00 │
│ errors_local          │ 22.83 KiB │ 0000-00-00 │ 0000-00-00 │
│ outcomes_raw_local    │ 2.59 KiB  │ 0000-00-00 │ 0000-00-00 │
│ outcomes_hourly_local │ 1.81 KiB  │ 0000-00-00 │ 0000-00-00 │
└───────────────────────┴───────────┴────────────┴────────────┘
```
(This was the query used: https://gist.github.com/sanchezzzhak/511fd140e8809857f8f1d84ddb937015)

Here is migrations on clickhouse:
```
┌─group────────┬─migration_id────────────────────────────────────────────┬───────────timestamp─┬─status────┬─version─┐
│ discover     │ 0001_discover_merge_table                               │ 2021-05-14 10:22:30 │ completed │       2 │
│ discover     │ 0002_discover_add_deleted_tags_hash_map                 │ 2021-05-14 10:22:30 │ completed │       2 │
│ discover     │ 0003_discover_fix_user_column                           │ 2021-05-14 10:22:30 │ completed │       2 │
│ discover     │ 0004_discover_fix_title_and_message                     │ 2021-05-14 10:22:30 │ completed │       2 │
│ discover     │ 0005_discover_fix_transaction_name                      │ 2021-05-14 10:22:30 │ completed │       2 │
│ discover     │ 0006_discover_add_trace_id                              │ 2021-05-14 10:22:30 │ completed │       2 │
│ events       │ 0001_events_initial                                     │ 2021-05-14 10:22:29 │ completed │       2 │
│ events       │ 0002_events_onpremise_compatibility                     │ 2021-05-14 10:22:29 │ completed │       2 │
│ events       │ 0003_errors                                             │ 2021-05-14 10:22:29 │ completed │       2 │
│ events       │ 0004_errors_onpremise_compatibility                     │ 2021-05-14 10:22:29 │ completed │       2 │
│ events       │ 0005_events_tags_hash_map                               │ 2021-05-14 10:22:29 │ completed │       2 │
│ events       │ 0006_errors_tags_hash_map                               │ 2021-05-14 10:22:29 │ completed │       2 │
│ events       │ 0007_groupedmessages                                    │ 2021-05-14 10:22:29 │ completed │       2 │
│ events       │ 0008_groupassignees                                     │ 2021-05-14 10:22:29 │ completed │       2 │
│ events       │ 0009_errors_add_http_fields                             │ 2021-05-14 10:22:30 │ completed │       2 │
│ events       │ 0010_groupedmessages_onpremise_compatibility            │ 2021-05-14 10:22:30 │ completed │       2 │
│ events       │ 0011_rebuild_errors                                     │ 2021-05-14 10:22:30 │ completed │       2 │
│ events       │ 0012_errors_make_level_nullable                         │ 2021-05-14 10:22:30 │ completed │       2 │
│ events       │ 0013_errors_add_hierarchical_hashes                     │ 2021-05-14 10:22:30 │ completed │       2 │
│ events       │ 0014_backfill_errors                                    │ 2021-05-14 10:22:30 │ completed │       2 │
│ events       │ 0015_truncate_events                                    │ 2021-05-14 10:22:30 │ completed │       2 │
│ outcomes     │ 0001_outcomes                                           │ 2021-05-14 10:22:30 │ completed │       2 │
│ outcomes     │ 0002_outcomes_remove_size_and_bytes                     │ 2021-05-14 10:22:30 │ completed │       2 │
│ outcomes     │ 0003_outcomes_add_category_and_quantity                 │ 2021-05-14 10:22:30 │ completed │       2 │
│ outcomes     │ 0004_outcomes_matview_additions                         │ 2021-05-14 10:22:30 │ completed │       2 │
│ sessions     │ 0001_sessions                                           │ 2021-05-14 10:22:30 │ completed │       2 │
│ sessions     │ 0002_sessions_aggregates                                │ 2021-05-14 10:22:30 │ completed │       2 │
│ sessions     │ 0003_sessions_matview                                   │ 2021-05-14 10:22:30 │ completed │       2 │
│ system       │ 0001_migrations                                         │ 2021-05-14 10:22:29 │ completed │       1 │
│ transactions │ 0001_transactions                                       │ 2021-05-14 10:22:30 │ completed │       2 │
│ transactions │ 0002_transactions_onpremise_fix_orderby_and_partitionby │ 2021-05-14 10:22:30 │ completed │       2 │
│ transactions │ 0003_transactions_onpremise_fix_columns                 │ 2021-05-14 10:22:30 │ completed │       2 │
│ transactions │ 0004_transactions_add_tags_hash_map                     │ 2021-05-14 10:22:30 │ completed │       2 │
│ transactions │ 0005_transactions_add_measurements                      │ 2021-05-14 10:22:30 │ completed │       2 │
│ transactions │ 0006_transactions_add_http_fields                       │ 2021-05-14 10:22:30 │ completed │       2 │
│ transactions │ 0007_transactions_add_discover_cols                     │ 2021-05-14 10:22:30 │ completed │       2 │
│ transactions │ 0008_transactions_add_timestamp_index                   │ 2021-05-14 10:22:30 │ completed │       2 │
│ transactions │ 0009_transactions_fix_title_and_message                 │ 2021-05-14 10:22:30 │ completed │       2 │
│ transactions │ 0010_transactions_nullable_trace_id                     │ 2021-05-14 10:22:30 │ completed │       2 │
│ transactions │ 0011_transactions_add_span_op_breakdowns                │ 2021-05-14 10:22:30 │ completed │       2 │
└──────────────┴─────────────────────────────────────────────────────────┴─────────────────────┴───────────┴─────────┘
```

Here's install log from 21.3 (where events display correctly) to 21.4: 

```
▶ Parsing command line ...

▶ Setting up error handling ...

▶ Checking minimum requirements ...

▶ Creating volumes for persistent storage ...
Created sentry-clickhouse.
Created sentry-data.
Created sentry-kafka.
Created sentry-postgres.
Created sentry-redis.
Created sentry-symbolicator.
Created sentry-zookeeper.

▶ Ensuring files from examples ...
../sentry/sentry.conf.py already exists, skipped creation.
../sentry/config.yml already exists, skipped creation.
../symbolicator/config.yml already exists, skipped creation.
../sentry/requirements.txt already exists, skipped creation.

▶ Generating secret key ...

▶ Replacing TSDB ...

▶ Fetching and updating Docker images ...
--no-ansi option is deprecated and will be removed in future versions. Use `--ansi never` instead.
Some service image(s) must be built from source by running:
    docker-compose build sentry-cleanup snuba-transactions-cleanup snuba-cleanup symbolicator-cleanup
21.4.1: Pulling from getsentry/sentry
Digest: sha256:3a213d32539701290f8eb244d0c6e849fae16ebba846a24e413f185987d6d2b5
Status: Image is up to date for getsentry/sentry:21.4.1
docker.io/getsentry/sentry:21.4.1

▶ Building and tagging Docker images ...

--no-ansi option is deprecated and will be removed in future versions. Use `--ansi never` instead.
smtp uses an image, skipping
memcached uses an image, skipping
redis uses an image, skipping
postgres uses an image, skipping
zookeeper uses an image, skipping
kafka uses an image, skipping
clickhouse uses an image, skipping
geoipupdate uses an image, skipping
snuba-api uses an image, skipping
snuba-consumer uses an image, skipping
snuba-outcomes-consumer uses an image, skipping
snuba-sessions-consumer uses an image, skipping
snuba-transactions-consumer uses an image, skipping
snuba-replacer uses an image, skipping
snuba-subscription-consumer-events uses an image, skipping
snuba-subscription-consumer-transactions uses an image, skipping
symbolicator uses an image, skipping
web uses an image, skipping
cron uses an image, skipping
worker uses an image, skipping
ingest-consumer uses an image, skipping
post-process-forwarder uses an image, skipping
subscription-consumer-events uses an image, skipping
subscription-consumer-transactions uses an image, skipping
relay uses an image, skipping
nginx uses an image, skipping
Building snuba-cleanup
Sending build context to Docker daemon  3.072kB
Step 1/5 : ARG BASE_IMAGE
Step 2/5 : FROM ${BASE_IMAGE}
 ---> cda5760ce0b7
Step 3/5 : RUN apt-get update && apt-get install -y --no-install-recommends cron &&     rm -r /var/lib/apt/lists/*
 ---> Using cache
 ---> 272eb5a520b2
Step 4/5 : COPY entrypoint.sh /entrypoint.sh
 ---> Using cache
 ---> ca28fca02cd8
Step 5/5 : ENTRYPOINT ["/entrypoint.sh"]
 ---> Using cache
 ---> 78c5ff2ba5a0
Successfully built 78c5ff2ba5a0
Successfully tagged snuba-cleanup-onpremise-local:latest
Building snuba-transactions-cleanup
Sending build context to Docker daemon  3.072kB
Step 1/5 : ARG BASE_IMAGE
Step 2/5 : FROM ${BASE_IMAGE}
 ---> cda5760ce0b7
Step 3/5 : RUN apt-get update && apt-get install -y --no-install-recommends cron &&     rm -r /var/lib/apt/lists/*
 ---> Using cache
 ---> 272eb5a520b2
Step 4/5 : COPY entrypoint.sh /entrypoint.sh
 ---> Using cache
 ---> ca28fca02cd8
Step 5/5 : ENTRYPOINT ["/entrypoint.sh"]
 ---> Using cache
 ---> 78c5ff2ba5a0
Successfully built 78c5ff2ba5a0
Successfully tagged snuba-cleanup-onpremise-local:latest
Building symbolicator-cleanup
Sending build context to Docker daemon  3.072kB
Step 1/5 : ARG BASE_IMAGE
Step 2/5 : FROM ${BASE_IMAGE}
 ---> 77d674e404a3
Step 3/5 : RUN apt-get update && apt-get install -y --no-install-recommends cron &&     rm -r /var/lib/apt/lists/*
 ---> Using cache
 ---> a9cb3ad82f09
Step 4/5 : COPY entrypoint.sh /entrypoint.sh
 ---> Using cache
 ---> 33932114dc8e
Step 5/5 : ENTRYPOINT ["/entrypoint.sh"]
 ---> Using cache
 ---> ec1572283e2d
Successfully built ec1572283e2d
Successfully tagged symbolicator-cleanup-onpremise-local:latest
Building sentry-cleanup
Sending build context to Docker daemon  3.072kB
Step 1/5 : ARG BASE_IMAGE
Step 2/5 : FROM ${BASE_IMAGE}
 ---> e2ea6046c3c1
Step 3/5 : RUN apt-get update && apt-get install -y --no-install-recommends cron &&     rm -r /var/lib/apt/lists/*
 ---> Using cache
 ---> 21c6d96062c9
Step 4/5 : COPY entrypoint.sh /entrypoint.sh
 ---> Using cache
 ---> 29047db6ef60
Step 5/5 : ENTRYPOINT ["/entrypoint.sh"]
 ---> Using cache
 ---> 1d9a2fae543a
Successfully built 1d9a2fae543a
Successfully tagged sentry-cleanup-onpremise-local:latest

Docker images built.

▶ Turning things off ...
--no-ansi option is deprecated and will be removed in future versions. Use `--ansi never` instead.
Removing network onpremise_default
Network onpremise_default not found.
--no-ansi option is deprecated and will be removed in future versions. Use `--ansi never` instead.
Removing network sentry_onpremise_default
Network sentry_onpremise_default not found.

▶ Setting up Zookeeper ...
--no-ansi option is deprecated and will be removed in future versions. Use `--ansi never` instead.
Creating network "sentry_onpremise_default" with the default driver
Creating sentry_onpremise_zookeeper_run ...
Creating sentry_onpremise_zookeeper_run ... done
--no-ansi option is deprecated and will be removed in future versions. Use `--ansi never` instead.
Creating sentry_onpremise_zookeeper_run ...
Creating sentry_onpremise_zookeeper_run ... done
--no-ansi option is deprecated and will be removed in future versions. Use `--ansi never` instead.
Creating sentry_onpremise_zookeeper_run ...
Creating sentry_onpremise_zookeeper_run ... done

▶ Bootstrapping and migrating Snuba ...
--no-ansi option is deprecated and will be removed in future versions. Use `--ansi never` instead.
Creating sentry_onpremise_zookeeper_1 ...
Creating sentry_onpremise_redis_1     ...
Creating sentry_onpremise_clickhouse_1 ...
Creating sentry_onpremise_redis_1      ... done
Creating sentry_onpremise_zookeeper_1  ... done
Creating sentry_onpremise_kafka_1      ...
Creating sentry_onpremise_kafka_1      ... done
Creating sentry_onpremise_clickhouse_1 ... done
Creating sentry_onpremise_snuba-api_run ...
Creating sentry_onpremise_snuba-api_run ... done
2021-05-25 08:35:04,576 Attempting to connect to Kafka (attempt 0)...
2021-05-25 08:35:06,585 Attempting to connect to Kafka (attempt 1)...
2021-05-25 08:35:06,596 Connected to Kafka on attempt 1
2021-05-25 08:35:06,690 Creating Kafka topics...
--no-ansi option is deprecated and will be removed in future versions. Use `--ansi never` instead.
Creating sentry_onpremise_snuba-api_run ...
Creating sentry_onpremise_snuba-api_run ... done
Finished running migrations

▶ Creating additional Kafka topics ...

▶ Ensuring proper PostgreSQL version ...

▶ Setting up / migrating database ...
--no-ansi option is deprecated and will be removed in future versions. Use `--ansi never` instead.
Creating sentry_onpremise_symbolicator_1 ...
Creating sentry_onpremise_memcached_1    ...
Creating sentry_onpremise_postgres_1     ...
Creating sentry_onpremise_smtp_1         ...
Creating sentry_onpremise_snuba-subscription-consumer-events_1 ...
Creating sentry_onpremise_snuba-replacer_1                     ...
Creating sentry_onpremise_snuba-subscription-consumer-transactions_1 ...
Creating sentry_onpremise_snuba-consumer_1                           ...
Creating sentry_onpremise_snuba-api_1                                ...
Creating sentry_onpremise_snuba-transactions-consumer_1              ...
Creating sentry_onpremise_snuba-sessions-consumer_1                  ...
Creating sentry_onpremise_snuba-outcomes-consumer_1                  ...
Creating sentry_onpremise_memcached_1                                ... done
Creating sentry_onpremise_smtp_1                                     ... done
Creating sentry_onpremise_postgres_1                                 ... done
Creating sentry_onpremise_snuba-outcomes-consumer_1                  ... done
Creating sentry_onpremise_symbolicator_1                             ... done
Creating sentry_onpremise_snuba-api_1                                ... done
Creating sentry_onpremise_snuba-subscription-consumer-events_1       ... done
Creating sentry_onpremise_snuba-sessions-consumer_1                  ... done
Creating sentry_onpremise_snuba-consumer_1                           ... done
Creating sentry_onpremise_snuba-subscription-consumer-transactions_1 ... done
Creating sentry_onpremise_snuba-replacer_1                           ... done
Creating sentry_onpremise_snuba-transactions-consumer_1              ... done
Creating sentry_onpremise_web_run                                    ...
Creating sentry_onpremise_web_run                                    ... done
08:35:28 [INFO] sentry.plugins.github: apps-not-configured
Operations to perform:
  Apply all migrations: admin, auth, contenttypes, jira_ac, nodestore, sentry, sessions, sites, social_auth
Running migrations:
User Options: 100% |#############################################| Time: 0:00:00
 OK
  Applying sentry.0186_add_externalactor... OK
Creating missing DSNs
Correcting Group.num_comments counter

▶ Migrating file storage ...

▶ Generating Relay credentials ...
../relay/config.yml already exists, skipped creation.

▶ Setting up GeoIP integration ...
Setting up IP address geolocation ...
IP address geolocation database already exists.
IP address geolocation is not configured for updates.
See https://develop.sentry.dev/self-hosted/geolocation/ for instructions.
Error setting up IP address geolocation.


-----------------------------------------------------------------

You're all done! Run the following command to get Sentry running:

  docker-compose up -d

-----------------------------------------------------------------
```

Plenty of data still in Postgres Sentry database on 21.4:

```
        relation         | total_size
-------------------------+------------
 nodestore_node          | 1809 MB
 sentry_groupedmessage   | 12 MB
 sentry_groupemailthread | 4416 kB
 sentry_grouphash        | 1624 kB
 sentry_eventuser        | 1000 kB
(5 rows)
```

**Comment by BYK** [comment]
What is your upgrade process btw?

**Comment by cjd118** [comment]
I think we went:

9.1.2 > 10.0
10.0 > 20.12.1
20.12.1 > 21.3 (the last version we could get working)
21.3 > 21.4 / 21.5

Only volume that has been copied from 9.1.2 is the postgres volume. So we set up 10.0, ran install.sh. Then swapped out postgres volume with ours from 9.1.2, reran install.sh and modified docker-compose.yml to use postgres 11. Also copied config from 21.4 so we know it has all the settings.
